### PR TITLE
Switch from \cs_argument_spec:N to \cs_parameter_spec:N

### PR DESCRIPTION
**Pull request by josephwright**
**READ ME FIRST**: Please understand that in most cases we will not be able to merge a pull request because there are a lot of internal activities needed when updating the LaTeX2e sources. If you have a code suggestion please discuss it with the team first.

*Pull requests in this repository are intended for LaTeX Team members only.*

# Internal housekeeping

## Status of pull request

- Feedback wanted 
- Under development
- [x] Ready to merge

## Checklist of required changes before merge will be approved
- [ ] Test file(s) added N/A
- [x] Version and date string updated in changed source files
- [x] Relevant `\changes` entries in source included
- [x] Relevant `changes.txt` updated
- [ ] Rollback provided (if necessary)? N/A
- [ ] `ltnewsX.tex` (and/or `latexchanges.tex`) updated N/A

### CODE REVIEW

File: base/ltexpl.dtx [hunk 1]
```diff
@@ -356,9 +356,19 @@
...
+%         {Update name of \pkg{expl3} function}
+% \changes{v1.3h}{2024/04/17}
+%         {Add a kernel-level copy of \cs{cs_parameter_spec:N}}
 %    \begin{macrocode}
 \cs_gset_eq:NN \@expl@cs@prefix@spec@@N \cs_prefix_spec:N
-\cs_gset_eq:NN \@expl@cs@argument@spec@@N \cs_argument_spec:N
+\cs_if_exist:NTF \cs_parameter_spec:N
+  { \cs_gset_eq:NN \@expl@cs@parameter@spec@@N \cs_parameter_spec:N }
+  { \cs_gset_eq:NN \@expl@cs@parameter@spec@@N \cs_argument_spec:N }
+\cs_gset_eq:NN \__kernel_cs_parameter_spec:N \@expl@cs@parameter@spec@@N
```
Comment (muzimuzhi) [inline base/ltexpl.dtx:371]: Does the new `\__kernel_cs_parameter_spec:N` need to be undefined in rollback code?

File: base/changes.txt [hunk 2]
```diff
@@ -6,6 +6,12 @@ to completeness or accuracy and it contains some references to files that are
 not part of the distribution.
 ================================================================================
 
+2024-04-17 Joseph Wright  <Joseph.Wright@latex-project.org>
+	* ltexpl.dtx, ltdefns.dtx
+	Rename \@expl@cs@argument@spec@@N to \@expl@cs@parameter@spec@@N (gh/1014)
+	* ltexpl.dtx, ltcmd.dtx, ltcmdhooks.dtx
+	Use \cs_parameter_spec:N in place of \cs_argument_spec:N (gh/1014)
```
Comment (muzimuzhi) [inline base/changes.txt:13]: Mention `\__kernel_cs_parameter_spec:N` too?

File: base/ltexpl.dtx (hunk 1 above)
Comment (josephwright) [reply base/ltexpl.dtx:371]: I don't think so - this is strictly kernel-internl

File: base/changes.txt (hunk 2 above)
Comment (josephwright) [reply base/changes.txt:13]: I don't think the changes needs to go to that level of detail myself

Comment (FrankMittelbach) [reply base/changes.txt:13]: We don't normally roll back internals unless there is a very good reason. Being internal they should not confliced with anything if they stay around.

Comment (FrankMittelbach) [reply base/changes.txt:13]: ups comment in the wrong place :-)

File: base/ltexpl.dtx (hunk 1 above)
Comment (muzimuzhi) [reply base/ltexpl.dtx:371]: I don't insist on adding it, but is `\__kernel_cs_parameter_spec:N` more kernel-internal than `\@expl@cs@parameter@spec@@N`? The latter is undefined in rollback code.

Comment (FrankMittelbach) [reply base/ltexpl.dtx:371]: We usually undefine in rollback if there is a real chance that the new command was used in the past in documents or code. That really isn't the case for `\@expl@cs@parameter@spec@@N` so I gues it doesn't have to be undefined.  -- Guess it doesn't hurt either way

**Review by FrankMittelbach — APPROVED** [review]

**Pull request merged**